src/database/sqlite/sqlite_metadata.c

```diff
@@ -2337,6 +2340,62 @@ uint64_t sqlite_get_meta_space(void)
     return sqlite_get_db_space(db_meta);
 }
 
+#define SQL_ADD_AGENT_EVENT_LOG      \
+    "INSERT INTO agent_event_log (event_type, version, value, date_created) VALUES " \
+    " (@event_type, @version, @value, UNIXEPOCH())"
+
+void add_agent_event(event_log_type_t event_id, int64_t value)
+{
+    sqlite3_stmt *res = NULL;
+
+    if (!PREPARE_STATEMENT(db_meta, SQL_ADD_AGENT_EVENT_LOG, &res))
+        return;
+
+    int param = 0;
+    SQLITE_BIND_FAIL(done, sqlite3_bind_int(res, ++param, event_id));
+    SQLITE_BIND_FAIL(done, sqlite3_bind_text(res, ++param, NETDATA_VERSION, -1, SQLITE_STATIC));
+    SQLITE_BIND_FAIL(done, sqlite3_bind_int64(res, ++param, value));
+
+    param = 0;
+    int rc = execute_insert(res);
+    if (rc != SQLITE_DONE)
+        error_report("Failed to store agent event information, rc = %d", rc);
+done:
+    REPORT_BIND_FAIL(res, param);
+    SQLITE_FINALIZE(res);
+}
+
+void cleanup_agent_event_log(void)
+{
+    db_execute(db_meta, "DELETE FROM agent_event_log WHERE date_created < UNIXEPOCH() - 30 * 86400");
+}
+
+#define SQL_GET_AGENT_EVENT_TYPE_MEDIAN                                                                                \
+    "SELECT AVG(value) AS median FROM "                                                                                \
```

Comment:
This is a separate module that we don't currently include (maybe in the future)

What this query does is calculate the median by ordering the data, then go the middle and get the value.
For odd number of values it can pick the one directly. For even number of values it will get the average of the two in the middle.

This is why there is an avg function there -- it will be either one value or average of two (but it is the median) -- the `as median` just names that result column `median`